src/katgpucbf/fgpu/engine.py

```diff
@@ -377,11 +377,30 @@ def capacity(self) -> int:  # noqa: D401
         # basically, the number of spectra.
         return self.spectra.shape[0] * self.spectra.shape[2]
 
+    @property
+    def next_timestamp(self) -> int:  # noqa: D401
```

Comment:
Take a look for where `max_end_out` is assigned: you can use this property to make this simplification:
```diff
-            max_end_out = self._out_item.timestamp + self._out_item.capacity * self.output.spectra_samples
+            max_end_out = self._out_item.next_timestamp